extensions/amp-user-notification/0.1/amp-user-notification.js

```diff
@@ -209,20 +214,42 @@ export class AmpUserNotification extends AMP.BaseElement {
    * @return {!Promise}
    */
   postDismissEnpoint_() {
+    const enctype = this.element.getAttribute('enctype') ||
+      'application/json;charset=utf-8';
     return Services.xhrFor(this.win).fetchJson(
         dev().assertString(this.dismissHref_),
-        {
-          method: 'POST',
-          credentials: 'include',
-          requireAmpResponseSourceOrigin: false,
-          body: /** @type {!JsonObject} */({
-            'elementId': this.elementId_,
-            'ampUserId': this.ampUserId_,
-          }),
-          headers: {
-            'Content-Type': 'application/json;charset=utf-8',
-          },
-        });
+        this.buildPostDismissRequest_(enctype, {
+          elementId: this.elementId_,
+          ampUserId: this.ampUserId_,
+        }));
+  }
+
+  /**
+   * Creates a Request to be used for postDismiss
+   * @private
+   * @param {!string} enctype
+   * @param {!Object} parameters
```

Comment:
It should be a inlined typedef in closure:

```
@param {!{elementId: string, ampUserId: string}} parameters
```

Reply:
tried doing this, but TRAVIS threw error:

https://travis-ci.org/ampproject/amphtml/jobs/282682341

Reply:
Hmm. And before it wasn't throwing errors? You just need to replace args `{...}` with `parameters` since you've already defined what's in that structure in the `@typedef`.